### DISCUSSION TITLE
Restricting to Route/Vehicle Types for searches

### DISCUSSION THREAD

**Discussion by cfoehrdes** [motis-project/motis#397] [Q&A] [answered]
Is there a way I can implement some restriction to the routes/vehicles. I found `search_options` with a profile but not sure how a profile is defined and if it is what I am looking for.

One example would be to restrict a search on Germany data to transportation covered by the national "49€" ticket. Another example could be avoiding long distance buses?

**Comment by felixguendling** [comment, marked as answer]
It would probably be possible, to exclude those routes in [the routing algorithm](https://github.com/motis-project/nigiri/blob/master/include/nigiri/routing/raptor/raptor.h). For 100% correctness, also the "route" definition of the RAPTOR algorithm would need to be adjusted so that all transports that are grouped in one route have the same properties regarding all filter search options (i.e. all are 49€ ticket compatible or non of them are). In most cases, this would probably be the case anyway so as a first step, it would be sufficient to store a boolean for each transport "is 49€ compatible" and filter routes that are not compatible. Same for long distance buses.

Another option (that would not require to change the algorithm and data loader) would be to filter the data and create a GTFS dataset that does not contain the trips that are not eligible for the 49€ ticket (same for long distance buses). Then start a separate server with the filtered data. Since in most cases you'll have a load balancer anyway, the load balancer can then redirect the queries to the server with the filtered/unfiltered dataset depending on the search options.

**Reply by cfoehrdes** [reply]
Thank you for the explanation! I also thought about you second approach, cause I don't want to touch the code. I think on the long term it would be great to have some sort of profile you can pass on search query which filters some routes by rule or similar.
